src/python/library/tritonclient/utils/cuda_shared_memory/cuda_shared_memory.cc

```diff
@@ -157,6 +157,37 @@ CudaSharedMemoryRegionSet(
   return 0;
 }
 
+int
+CudaSharedMemoryRegionSetDptr(
```

Comment:
Both CudaSharedMemoryRegionSet and CudaSharedMemoryRegionSetDptr are identical. Why not use a device flag and condition on that to use cudaMemcpyDeviceToDevice or cudaMemcpyHostToDevice?